ENIGMAsystem/SHELL/Platforms/xlib/XLIBwindow.cpp

```diff
@@ -267,15 +268,50 @@ bool window_get_stayontop() {
 }
 
 void window_set_sizeable(bool sizeable) {
-  if (enigma::isSizeable == sizeable) return;
   enigma::isSizeable = sizeable;
+  XSizeHints *sh = XAllocSizeHints();
+  sh->flags = PMinSize | PMaxSize;
+  
+  if (enigma::window_min_width == -1)
+    enigma::window_min_width = 1;
+  if (enigma::window_max_width == -1) 
+    enigma::window_max_width = display_get_width();
+  if (enigma::window_min_height == -1) 
+    enigma::window_min_height = 1;
+  if (enigma::window_max_height == -1) 
+    enigma::window_max_height = display_get_height();
+  
+  if (sizeable) {
+    sh->min_width = enigma::window_min_width;
+    sh->max_width = enigma::window_max_width;
+    sh->min_height = enigma::window_min_height;
+    sh->max_height = enigma::window_max_height;
+  } else {
+    sh->min_width = sh->max_width = window_get_width();
+    sh->min_height = sh->max_height = window_get_height();
+  }
+  XSetWMSizeHints(disp, win, sh, XA_WM_NORMAL_HINTS);
```

Comment:
```suggestion
  XSetWMNormalHints(disp, win, sh);
```

Reply:
I guess it doesn't really save any lines, just a parameter, but you may as well do it anyway.